### PR TITLE
feat: Adding new sample for the python callouts

### PR DESCRIPTION
**Pull request by pweiber**
Adding the redirect sample as a standalone example

## Use-cases
7. Perform a HTTP redirect

## Details
UC7 - For a given URL - redirect to another URL

### CODE REVIEW

File: callouts/python/extproc/tests/redirect_test.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+
+class CalloutServerTest(callout_server.CalloutServer):
+
+  def on_request_headers(
+      self, headers: HttpHeaders, context: ServicerContext
+  ) -> HeadersResponse:
+    """Custom processor on request headers."""
+    return callout_server.add_header_mutation(
+        add=[('host', 'service-extensions.com'), (':path', '/extensions')],
```
Comment (rshriram) [inline callouts/python/extproc/tests/redirect_test.py:48]: dont you have to set the immediate response field for redirect?

Comment (pweiber) [reply callouts/python/extproc/tests/redirect_test.py:48]: I was able to redirect without it, using just like this, I called /callouts for example and it redirected to the /extensions

Comment (rshriram) [reply callouts/python/extproc/tests/redirect_test.py:48]: sure that is for the same host. But in that case, it is not a redirect. Its a rewrite. The browser's address bar does not change. A redirect requires sending HTTP 301

Comment (pweiber) [reply callouts/python/extproc/tests/redirect_test.py:48]: Hmmm got it, I'll address the changes then

**Pull request closed without merging**